p2p/exchange_test.go

```diff
@@ -156,6 +157,70 @@ func TestExchange_RequestHead_UnresponsivePeer(t *testing.T) {
 	assert.NotNil(t, head)
 }
 
+func TestExchange_RequestHeadFlightProtection(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	hosts := createMocknet(t, 3)
+	exchg, trustedStore := createP2PExAndServer(t, hosts[0], hosts[1])
+
+	// create the same requests
+	tests := []struct {
+		requestFromTrusted bool
+		lastHeader         *headertest.DummyHeader
+		expectedHeight     uint64
+		expectedHash       header.Hash
+	}{
+		{
+			requestFromTrusted: true,
+			lastHeader:         trustedStore.Headers[trustedStore.HeadHeight-1],
+			expectedHeight:     trustedStore.HeadHeight,
+			expectedHash:       trustedStore.Headers[trustedStore.HeadHeight].Hash(),
+		},
+		{
+			requestFromTrusted: true,
+			lastHeader:         trustedStore.Headers[trustedStore.HeadHeight-1],
+			expectedHeight:     trustedStore.HeadHeight,
+			expectedHash:       trustedStore.Headers[trustedStore.HeadHeight].Hash(),
+		},
```

Comment:
what's the difference between these two test cases?

Reply:
The duplicate cases serve to double-check that `singleflight` consistently returns the same response for identical requests from a `trusted` source. However, if you think a single case is sufficient for validation, we can remove the duplicates to streamline the test

Reply:
That seems like a good test generally-speaking but I don't think it is relevant to the single-flight protection feature and should rather be implemented in a different way. I appreciate the thought here, however.

Reply:
Got it, I’ll remove the test as it’s not directly relevant to single-flight. Thanks for the feedback!